components/templates/templates.js

```diff
@@ -330,5 +337,27 @@ if (!utils.templates) {
         target.appendChild(templates.item(c));
       }
     };
+
+    if (document.registerElement) {
+      var xTemplateProto = Object.create(HTMLTemplateElement.prototype);
+
+      // When an 'x-template' is created the element in the DOM is destroyed and
+      // moved to the Shadow DOM
+      xTemplateProto.createdCallback = function() {
```

Comment:
I think this code is executed for <template is="x-template"> nodes added to DOM after parsing this library, right? So we could add a description for third parties